worbee/lib/worbee/core.ex

```diff
@@ -20,34 +20,48 @@ defmodule Worbee.Core do
     game.guesses
   end
 
-  def show_results(%__MODULE__{guesses: []}) do
+  def compute_latest_guess(%__MODULE__{guesses: []}) do
     []
   end
 
-  # convert
-  def show_results(game) do
-    [latest_guess | _rest] = game.guesses
+  def compute_latest_guess(%{answer: answer, guesses: [latest_guess | _]} = _game) do
+    guess_graphemes = String.graphemes(latest_guess)
+    answer_graphemes = String.graphemes(answer)
 
-    latest_guess
-    |> String.split("", trim: true)
-    |> Enum.with_index()
-    |> Enum.map(fn {term, index} -> {term, char_comparison(term, index, game.answer)} end)
+    wrongs = guess_graphemes -- answer_graphemes
+
+    compute_greens(guess_graphemes, answer_graphemes)
+    |> compute_grays(wrongs)
+    |> compute_yellows()
+    |> convert_keys_to_atoms()
   end
 
-  # :correct, :wrong, :close
-  defp char_comparison(guess_char, index, answer) do
-    current_char = String.at(answer, index)
+  defp compute_greens(guess_graphemes, answer_graphemes) do
```

Comment:
After zipping both lists of graphemes, any elements that are in the same position and are equal are considered correct (green).